docs/proposals/metric-tab-and-metric-puller.md

```diff
@@ -0,0 +1,134 @@
+---
+title: Metric T
+authors:
+  - `@sarabala1979` and  `dsmithco` # Authors' github accounts here.
+sponsors:
+  - TBD        # List all interested parties here.
+reviewers:
+  - "@alexmt"
+  - TBD
+approvers:
+  - "@alexmt"
+  - TBD
+
+creation-date: 2022-06-13
+last-updated: 2022-06-13
+---
+
+## Summary
+
+Enhancing ArgoCD UI to display the Resource Metrics on the Resource Page will help the user to quickly triage 
+the resource status and make decisions instead of viewing other metrics tools.
+ 
+
+## Motivation
+
+Currently, ArgoCD user needs to go to third-party tools to view the resource KPI for triaging the issue which is 
+time-consuming to decide on application recovery.
+
+### Goals
+1. Show the Metrics tab in Resource tab
+2. Show the configured dashboard in Metrics tab
+3. Able to configure the dashboard for each application and resources
+4. Support Prometheus as metrics provide in MVP release
+
+
+### Non-Goals
+
+1. Adding Caching layer to improve the API performance
+
+## Proposal
+
+#####ArgoCD UI:
+Enhancing ArgoCD UI to support the Metrics tab on the resource page. This Tab will be enabled and disabled using the feature flag. 
+This Metrics component will make an API call to render the metrics dashboard from the new Metrics Server.
+
+#####ArgoCD Server:
+Enhancing ArgoCD Server to have metrics puller and will expose the APIs to provide the Dashboard and Metrics information. Metrics Puller will read the Dashboard, Graph, and metrics provider configuration from Configmap.  
+
+#####Authentication:
+No need for separate implementation
+
+#####Defining Metrics page:
+The metrics page is fully configurable. UI will call API to get the dashboard layout and graph information from the server. User
+
+#####Enable/Disable the feature:
+The server will return HTTP resource code:405  if the feature is not enabled. UI will show/hide the metrics tab.
+
+
+### Use cases
+1. User can compare KPIs for rollout releases  quickly.
+2. User can review the service performance.
+
+### Implementation Details/Notes/Constraints [optional]
+
+##### Metrics Tap:
+This feature is going to be first class citizen in ArgoCDUI. Adding Metrics Component in Resource page. Implement Metrics
+Service  to make API call to query the Dashboard configuration and Graph data.
+
+#####Metrics Puller:
+Metrics Puller module will be implemented in ArgoCD Server. ArgoCD API server to expose the APIs for Metrics Dashboard and Graph data.
```

Comment:
Please add a section about how performance issues will be mitigated to avoid impacting ArgoCD API server.